addons/google_account/google_account.py

```diff
@@ -34,30 +34,32 @@ def generate_refresh_token(self, cr, uid, service, authorization_code, context=N
             content = urllib2.urlopen(req).read()
         except urllib2.HTTPError:
             error_msg = "Something went wrong during your token generation. Maybe your Authorization Code is invalid or already expired"
-            raise self.pool.get('res.config.settings').get_config_warning(cr, _(error_msg), context=context)
+            raise self.env['res.config.settings'].get_config_warning(_(error_msg))
 
         content = simplejson.loads(content)
         return content.get('refresh_token')
 
-    def _get_google_token_uri(self, cr, uid, service, scope, context=None):
-        ir_config = self.pool['ir.config_parameter']
+    @api.model
+    def _get_google_token_uri(self, service, scope):
+        ir_config = self.env['ir.config_parameter']
         params = {
             'scope': scope,
-            'redirect_uri': ir_config.get_param(cr, SUPERUSER_ID, 'google_redirect_uri'),
-            'client_id': ir_config.get_param(cr, SUPERUSER_ID, 'google_%s_client_id' % service),
+            'redirect_uri': ir_config.sudo().get_param('google_redirect_uri'),
+            'client_id': ir_config.sudo().get_param('google_%s_client_id' % service),
             'response_type': 'code',
-            'client_id': ir_config.get_param(cr, SUPERUSER_ID, 'google_%s_client_id' % service),
+            'client_id': ir_config.sudo().get_param('google_%s_client_id' % service),
         }
         uri = 'https://accounts.google.com/o/oauth2/auth?%s' % werkzeug.url_encode(params)
         return uri
 
     # If no scope is passed, we use service by default to get a default scope
-    def _get_authorize_uri(self, cr, uid, from_url, service, scope=False, context=None):
+    @api.model
+    def _get_authorize_uri(self,from_url, service, scope=False):
         """ This method return the url needed to allow this instance of OpenErp to access to the scope of gmail specified as parameters """
-        state_obj = dict(d=cr.dbname, s=service, f=from_url)
+        state_obj = dict(d=self._cr.dbname, s=service, f=from_url)
```

Comment:
* use dict literal
* get cursor via env, `self._cr` is a backwards-compatibility alias and may not remain eternally